src/webhook/upgrade_operator/fetch_upgradeconfig.py

```diff
@@ -0,0 +1,40 @@
+from kubernetes import client, config
+from kubernetes.client.rest import ApiException
+from kubernetes.config.config_exception import ConfigException
+
+
+def kube_auth():
+    # Use ~/.kube/config for authentication if script run directly
+    if __name__ == "__main__":
+        try:
+            config.load_kube_config()
+            return True
+        except ConfigException as c:
+            print("ConfigException in load_kube_config() : %s\n" % c)
+            return False
+
+    # Use serviceaccount token to authenticate.
+    else:
+        try:
+            config.load_incluster_config()
+            return True
+        except ConfigException as c:
+            print("ConfigException in load_incluster_config() : %s\n" % c)
+            return False
+
+
+def get_upgradeconfig_cr():
+    if kube_auth():
```

Comment:
I'm noticing that the result of any exception is a printed message (which would go into the container logs somewhere?) and a `None` return (which is sometimes implicit, e.g. in the case of failed auth). That may be okay depending on how this is supposed to be used by its consuming code. But it's an esoteric enough pattern that it would be nice to have it spelled out in a docstring.